instructor/ui/instructor.js

```diff
@@ -489,10 +474,80 @@ $(function() {
       paper.view.update();
 
     });
+
     load_sketch_tool(x_upper_bound, tick_values);
+
+
+    mypapers[3].setup("noise-canvas");
+
+    /*
+      Preview noise for sketched graph
+     */
+    $('.scb_ab_f_add_noise').click(function () {
+      /* Can preview noise, when graph has been plotted */
+      if (typeof(data_points) != 'undefined' && data_points.length > 0) {
+        if ($(".scb_ab_s_preview_canvas_div").css('display') == 'none') {
+          $(".scb_ab_s_preview_canvas_div").show();
+          $(".scb_ab_f_add_noise").text('CLOSE');
+          mypapers[3].activate();
+          paper.project.clear();
+          paper.project.view.viewSize.width = 450;
+          paper.project.view.viewSize.height = 250;
+          var fraction = 450.0 / 600;
+          var y_fraction = 250.0 / 300;
+          /* x axis */
+          draw_line(X_ORIGIN * fraction, Y_ORIGIN * y_fraction, X_AXIS_LAST_VALUE * fraction, Y_ORIGIN * y_fraction);
+          /* y axis */
+          draw_line(
+            X_ORIGIN * fraction,
+            (Y_ORIGIN - Y_AXIS_LENGTH_PX) * y_fraction,
+            X_ORIGIN * fraction,
+            Y_ORIGIN * y_fraction
+          );
+          var path = new Path();
+          path.strokeColor = 'black';
+          path.strokeWidth = '1';
+          var factor = 0.05;
+          var regular_points = getDataPoints();
+          _.each(data_points, function (point, i) {
+            path.add(new Point(
+              point[0] * fraction,
+              point[1] * y_fraction + regular_points[i][1]*(factor + 2 * factor * Math.random())))
```

Comment:
Can you refactor this noise generation into its own function? Something like this would work: `var dataWithNoise = addNoise(getDataPoints())`. It would also be nice to reuse that function in the noise generation in `FACSModelFactory.js`, not sure how difficult that is though